File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.java
class Solution {
    public int minOperations(int[][] grid, int x) {
        int[] arr = new int[grid.length * grid[0].length];
        int remainder = grid[0][0] % x;
        int operations = 0;

        for (int i = 0; i < grid.length; i++) {
            for (int j = 0; j < grid[0].length; j++) {
                if (grid[i][j] % x != remainder)
                    return -1;

                arr[i * grid[0].length + j] = grid[i][j];
            }
        }

        Arrays.sort(arr);
        int median = arr[(grid.length * grid[0].length) / 2];

        for (int num : arr) {
            operations += Math.abs(num - median) / x;
        }

        return operations;
    }
}

File: 2160-minimum-operations-to-make-a-uni-value-grid/minimum-operations-to-make-a-uni-value-grid.go
func minOperations(grid [][]int, x int) int {
    m, n := len(grid), len(grid[0]);
	size := m * n;
	arr := make([]int, size);
	remainder := grid[0][0] % x;

    for i := 0; i < m; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j]%x != remainder {
				return -1;
			}

			arr[i*n+j] = grid[i][j];
		}
	}

    sort.Ints(arr);
	median := arr[size/2];
    operations := 0;

	for _, num := range arr {
		operations += abs(num - median) / x;
	}

	return operations;
}

func abs(a int) int {
	if a < 0 {
		return -a;
	}

	return a;
}